
package soap.jaxws.example.arun.com;

import javax.jws.WebMethod;
import javax.jws.WebParam;
import javax.jws.WebResult;
import javax.jws.WebService;
import javax.xml.bind.annotation.XmlSeeAlso;
import javax.xml.ws.Action;
import javax.xml.ws.FaultAction;
import javax.xml.ws.RequestWrapper;
import javax.xml.ws.ResponseWrapper;


/**
 * This class was generated by the JAX-WS RI.
 * JAX-WS RI 2.2.9-b130926.1035
 * Generated source version: 2.2
 * 
 */
@WebService(name = "personService", targetNamespace = "com.arun.example.jaxws.soap")
@XmlSeeAlso({
    ObjectFactory.class
})
public interface PersonService {


    /**
     * 
     * @param arg0
     * @return
     *     returns soap.jaxws.example.arun.com.Person
     * @throws TestException_Exception
     */
    @WebMethod
    @WebResult(targetNamespace = "")
    @RequestWrapper(localName = "getPerson", targetNamespace = "com.arun.example.jaxws.soap", className = "soap.jaxws.example.arun.com.GetPerson")
    @ResponseWrapper(localName = "getPersonResponse", targetNamespace = "com.arun.example.jaxws.soap", className = "soap.jaxws.example.arun.com.GetPersonResponse")
    @Action(input = "com.arun.example.jaxws.soap/personService/getPersonRequest", output = "com.arun.example.jaxws.soap/personService/getPersonResponse", fault = {
        @FaultAction(className = TestException_Exception.class, value = "com.arun.example.jaxws.soap/personService/getPerson/Fault/TestException")
    })
    public Person getPerson(
        @WebParam(name = "arg0", targetNamespace = "")
        Integer arg0)
        throws TestException_Exception
    ;

}
